src/daft-core/src/python/field.rs

```diff
@@ -38,22 +24,10 @@ impl PyField {
     pub fn eq(&self, other: &PyField) -> PyResult<bool> {
         Ok(self.field.eq(&other.field))
     }
-
-    pub fn __setstate__(&mut self, py: Python, state: PyObject) -> PyResult<()> {
-        match state.extract::<&PyBytes>(py) {
-            Ok(s) => {
-                self.field = bincode::deserialize(s.as_bytes()).unwrap();
-                Ok(())
-            }
-            Err(e) => Err(e),
-        }
-    }
-
-    pub fn __getstate__(&self, py: Python) -> PyResult<PyObject> {
-        Ok(PyBytes::new(py, &bincode::serialize(&self.field).unwrap()).to_object(py))
-    }
 }
 
+impl_bincode_py_state_serialization!(PyField);
```

Comment:
this might be a good macro to implement as a derive so we can `#[derive(PySerDe)]`